### PR TITLE
add java android demo

### LINKED ISSUE #1432: a java android demo
hey，sir.when i want to use sherp-onnx in our older project run by java, I found that the class OnlineRecognizer of the java api cannot use the constructor with assetmanager.So I recompiled the java api and wrote a java android demo.And could I submit a pull request?

### PR DESCRIPTION
**Pull request by JameWade**
Added an Android stream ASR demo written in Java
close #1432

### CODE REVIEW

File: android/SherpaOnnxJavaDemo/sherpa/src/main/java/com/k2fsa/sherpa/onnx/AudioEvent.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2024 Xiaomi Corporation
```
Comment (csukuangfj) [inline android/SherpaOnnxJavaDemo/sherpa/src/main/java/com/k2fsa/sherpa/onnx/AudioEvent.java]: Could you use symlinks?

You can create a symlink to the folder of
https://github.com/k2-fsa/sherpa-onnx/tree/master/sherpa-onnx/java-api/src

File: android/SherpaOnnxJavaDemo/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    package="com.k2fsa.sherpa.onnx">
+    <uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
+    <uses-permission android:name="android.permission.RECORD_AUDIO" />
+    <uses-permission android:name="android.permission.RECORD_AUDIO" />
+
+    <uses-permission android:name="android.permission.RECORD_AUDIO" />
+
+    <uses-permission android:name="android.permission.RECORD_AUDIO" />
+
```
Comment (csukuangfj) [inline android/SherpaOnnxJavaDemo/app/src/main/AndroidManifest.xml]: please keep only one and remove other 3.

File: android/SherpaOnnxJavaDemo/app/src/main/java/com/k2fsa/sherpa/onnx/OnlineRecognizer.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        String[] tokens = (String[]) arr[1];
+        float[] timestamps = (float[]) arr[2];
+        return new OnlineRecognizerResult(text, tokens, timestamps);
+    }
+
+
+    private native void delete(long ptr);
+
+    private native long newFromFile(OnlineRecognizerConfig config);
+    private native long newFromAsset(AssetManager assetManager,OnlineRecognizerConfig config);
```
Comment (csukuangfj) [inline android/SherpaOnnxJavaDemo/app/src/main/java/com/k2fsa/sherpa/onnx/OnlineRecognizer.java]:
```suggestion
    private native long newFromAsset(AssetManager assetManager, OnlineRecognizerConfig config);
```

Could you use Android studio to reformat your code/files ?

File: android/SherpaOnnxJavaDemo/app/src/main/java/com/k2fsa/sherpa/onnx/OnlineRecognizer.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
+// Copyright 2022-2023 by zhaoming
```
Comment (csukuangfj) [inline android/SherpaOnnxJavaDemo/app/src/main/java/com/k2fsa/sherpa/onnx/OnlineRecognizer.java:1]: By the way, can you use a symlink to
https://github.com/k2-fsa/sherpa-onnx/blob/master/sherpa-onnx/kotlin-api/OnlineRecognizer.kt
to replace this file and other files related to sherpa-onnx Java APIs?

I think you can invoke directly Kotlin APIs in Android Java.

File: android/SherpaOnnxJavaDemo/app/src/main/java/com/k2fsa/sherpa/onnx/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+
+
+    private void startSpeechService(){
+        // 启动语音识别服务
+        Intent serviceIntent = new Intent(this, SpeechSherpaRecognitionService.class);
+        ContextCompat.startForegroundService(this, serviceIntent); // 启动前台服务（如果需要）
+        appViewModel = new ViewModelProvider(Application.getInstance()).get(AppViewModel.class);
+        // 观察语音识别结果
```
Comment (csukuangfj) [inline android/SherpaOnnxJavaDemo/app/src/main/java/com/k2fsa/sherpa/onnx/MainActivity.java]: Could you remove the Chinese comments?

The code is self-explanatory and I think there is no need to add comments here.

File: android/SherpaOnnxJavaDemo/app/src/main/java/com/k2fsa/sherpa/onnx/Application.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    public static Application getInstance() {
+        return sApplication;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        sApplication = this;
+
+        //共享viewmodel
```
Comment (csukuangfj) [inline android/SherpaOnnxJavaDemo/app/src/main/java/com/k2fsa/sherpa/onnx/Application.java]: Please remove the comment.

File: android/SherpaOnnxJavaDemo/app/src/main/java/com/k2fsa/sherpa/onnx/OnlineRecognizer.java (hunk 3 above)
Comment (JameWade) [reply android/SherpaOnnxJavaDemo/app/src/main/java/com/k2fsa/sherpa/onnx/OnlineRecognizer.java]: Thank you for your suggestion. I have modified this part of the code according to your feedback.

**Review by csukuangfj — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**